File: src/fe/src/models/MaterialLaboratoryAnalysisInformation.ts
import { Gender } from "@/models/enums/Gender";
import { ShipmentMaterialCondition } from "@/models/enums/ShipmentMaterialCondition";
import { YesNoOption } from "./enums/YesNoOption";

export interface MaterialLaboratoryAnalysisInformation {
  Id: string;
  MaterialShippingInformationId: string;
  MaterialNumber: string;
  FreezingDate: Date;
  Temperature: number;
  UnitOfMeasureId: string;
  VirusConcentration: string;
  CulturingCellLine: string;
  CulturingPassagesNumber: number;
  CollectedSpecimenTypes: string[];
  TypeOfTransportMedium: string;
  BrandOfTransportMedium: string;
  GSDUploadedToDatabase: YesNoOption;
  DatabaseUsedForGSDUploadingId: string;
  AccessionNumberInGSDDatabase: string;
}

File: src/be/src/WHO.BioHub.Models/Models/MaterialLaboratoryAnalysisInformation.cs
using WHO.BioHub.Models.Models.Abstractions;
using WHO.BioHub.Shared.Enums;

namespace WHO.BioHub.Models.Models;

public class MaterialLaboratoryAnalysisInformation : EntityBase
{
    public Guid? MaterialShippingInformationId { get; set; }
    public virtual MaterialShippingInformation MaterialShippingInformation { get; set; }
    public string MaterialNumber { get; set; }
    public DateTime? FreezingDate { get; set; }
    public double? Temperature { get; set; }
    public Guid? UnitOfMeasureId { get; set; }
    public virtual TemperatureUnitOfMeasure UnitOfMeasure { get; set; }
    public string VirusConcentration { get; set; }
    public string CulturingCellLine { get; set; }
    public int? CulturingPassagesNumber { get; set; }

    public IEnumerable<CollectedSpecimenType> CollectedSpecimenTypes { get; set; }

    public string TypeOfTransportMedium { get; set; }
    public string BrandOfTransportMedium { get; set; }

    public YesNoOption? GSDUploadedToDatabase { get; set; }

    public Guid? DatabaseUsedForGSDUploadingId { get; set; }
    public virtual GeneticSequenceData DatabaseUsedForGSDUploading { get; set; }

    public string AccessionNumberInGSDDatabase { get; set; }    
}
